e2e-tests/gef/cypress/integration/return-to-maker-min.spec.js

```diff
@@ -0,0 +1,110 @@
+import relative from './relativeURL';
+import returnToMaker from './pages/return-to-maker';
+import CREDENTIALS from '../fixtures/credentials.json';
+import applicationDetails from './pages/application-details';
+import automaticCover from './pages/automatic-cover';
+import manualInclusion from './pages/manual-inclusion-questionnaire';
+import securityDetails from './pages/security-details';
+import applicationSubmission from './pages/application-submission';
+import applicationPreview from './pages/application-preview';
+
+let applicationId;
+
+context('Return to Maker', () => {
+  before(() => {
+    cy.reinsertMocks();
+    cy.apiLogin(CREDENTIALS.CHECKER)
+      .then((token) => token)
+      .then((token) => {
+        cy.apiFetchAllApplications(token);
+      })
+      .then(({ body }) => {
+        applicationId = body.items[2]._id;
+
+        cy.login(CREDENTIALS.MAKER);
+      });
+  });
+
+  describe('creates and submits min', () => {
+    beforeEach(() => {
```

Comment:
I'm not sure if you need `beforeEach` here because there is only one test inside this `describe` block

Reply:
I think it's needed - we were debugging earlier - the user flow is
- maker logs in, does stuff
- checker logs in, does stuff
- maker logs in, does stuff

Each beforeEach logs in as the relevant user. Without this or with a single `before(() ...)`, the E2E file just errors in cypress. It also only seems to like "preserve cookies" in a beforeEach

If there is better way that'd be great :D

Reply:
oh! interesting behaviour. no idea what the alternative is then :) so leave it as it is